### ISSUE TITLE
add header

### ISSUE THREAD

**Issue by bfd69** [PrivateBin/PrivateBin#980]
Hello
is it possible to add a header when POSTing a comment ?
i secured privatebin with authelia so only members of my active directory can post, while posted bins are accessible anonymously. the only  drawback is that comments are not allowed, in fact the post for bins and comments are the same from browsers point of view. if you could add a header when posting a comment so i could : 
1) differentiate between the two operations
2) and then authorise comments based on that header

, a very simple header is sufficient like for example 'operation: comment'

thanks a lot

**Comment by elrido** [comment]
There was a guide in the wiki that describes how to do this with nginx - I'm sure this can be done with other webservers as well: https://github.com/PrivateBin/PrivateBin/wiki/Restrict-upload-using-NGINX

**Comment by bfd69** [comment]
first thanks a lot for replying, and for the link too
the solution in the link above is for adding auth to privatebin, even if you want to get a paste you have to provide credential to nginx.
my solution is based on traefik and authelia (based on cookie too), only authenticated users can create Pastes and non authenticated users can only read pastes, the moment they click on new,etc... they are asked for authentication.
but like the solution in the link above, only authenticated users can post comments, and thats because nothing differentiate a post for Past and a comment:
from reverse proxies and webbrowsers point of view the two posts queries are the same. 

in short : what i want is to make non authenticated users can post comments but not past.

a simple header posted with the comment request would be sufficient, 
i hope it was clear and again thanks a lot for hearing me, 

best regards

**Comment by elrido** [comment]
You'd have to add that in the javascript, I guess somewhere around here-ish:
https://github.com/PrivateBin/PrivateBin/blob/afd84da6b2a35ce779be9a5ebd51d2e2ff0a54cb/js/privatebin.js#L4558-L4589

The `ajaxHeaders` could get extended when a `pasteid` key got set in the `data` property of the `ServerInteraction` class. Remember to [regenerate the SRI hashes after editing the javascript file](https://github.com/PrivateBin/PrivateBin/wiki/FAQ#how-to-make-privatebin-work-when-i-have-changed-some-javascript-files).
